crates/fuel-core/src/query/message.rs

```diff
@@ -205,13 +222,19 @@ pub fn message_proof<T: MessageProofData + ?Sized>(
         .into_api_result::<CompressedBlock, StorageError>()?
     {
         Some(t) => t.into_inner().0,
-        None => return Ok(None),
+        None => {
+            return Ok(MessageProofEvaluation::Unavailable(
+                MessageProofUnavailability::UnavailableCommitBlockHeader,
+            ))
+        }
     };
 
     let block_height = *commit_block_header.height();
     if block_height == 0u32.into() {
         // Cannot look beyond the genesis block
-        return Ok(None)
+        return Ok(MessageProofEvaluation::Unavailable(
+            MessageProofUnavailability::InvalidBlockHeaderHeight,
+        ));
```

Comment:
Not sure why it is not possible to see beyond the genesys block so I just copied and pasted the comment to the `InvalidBlockHeaderHeight` variant.